tests/functional/src/main/java/org/mule/tck/junit4/FunctionalTestCase.java

```diff
@@ -64,26 +66,49 @@ protected String getConfigResources()
         return null;
     }
 
-
     @Override
     protected ConfigurationBuilder getBuilder() throws Exception
     {
+        String domainConfig = getDomainConfig();
+        domainContext = domainConfig != null ? new DomainContextBuilder().setDomainConfig(domainConfig).build() : null;
+        SpringXmlConfigurationBuilder configurationBuilder;
         String configResources = getConfigResources();
         if (configResources != null)
         {
-            return new SpringXmlConfigurationBuilder(configResources);
+            configurationBuilder = new SpringXmlConfigurationBuilder(configResources);
         }
-        configResources = getConfigFile();
-        if (configResources != null)
+        else
         {
-            if (configResources.contains(","))
+            configResources = getConfigFile();
+            if (configResources != null)
+            {
+                if (configResources.contains(","))
+                {
+                    throw new RuntimeException("Do not use this method when the config is composed of several files. Use getConfigFiles method instead.");
+                }
+                configurationBuilder = new SpringXmlConfigurationBuilder(configResources);
+            }
+            else
             {
-                throw new RuntimeException("Do not use this method when the config is composed of several files. Use getConfigFiles method instead.");
+                String[] multipleConfigResources = getConfigFiles();
+                configurationBuilder = new SpringXmlConfigurationBuilder(multipleConfigResources);
             }
-            return new SpringXmlConfigurationBuilder(configResources);
         }
-        String[] multipleConfigResources = getConfigFiles();
-        return new SpringXmlConfigurationBuilder(multipleConfigResources);
+        if (domainContext != null)
+        {
+            configurationBuilder.setDomainContext(domainContext);
+        }
+        return configurationBuilder;
+    }
+
+    @Override
+    protected void doTearDownAfterMuleContextDispose() throws Exception
+    {
+        super.doTearDownAfterMuleContextDispose();
+        if (domainContext != null)
+        {
+            domainContext.dispose();
```

Comment:
I think the check is made since the method is run in another annotated with "After" but itself is annotated with "AfterClass", so it runs twice and the second time the context is most likely supposed to have been disposed already. I added the check though and made sure the domainContext is set to null at the end too.